src/cpp/src/speculative_decoding/speculative_decoding_impl.hpp

```diff
@@ -51,4 +64,86 @@ class ContinuousBatchingPipeline::SpeculativeDecodingImpl : public ContinuousBat
     SpeculativeDecodingMetrics get_speculative_decoding_metrics();
 };
 
+class ContinuousBatchingPipeline::EagleDecodingImpl : public ContinuousBatchingPipeline::SpeculativeDecodingImpl {
+public:
+    EagleDecodingImpl(const ov::genai::ModelDesc& main_model_desc, const ov::genai::ModelDesc& draft_model_desc, const std::vector<int>& hidden_layers_to_abstract);
+
+    std::vector<EncodedGenerationResult>
+    generate(const std::vector<ov::Tensor>& input_ids,
+             const std::vector<GenerationConfig>& sampling_params,
+             const StreamerVariant& streamer,
+             std::optional<std::vector<ov::Tensor>> token_type_ids = std::nullopt) override;
+
+    GenerationHandle add_request(uint64_t request_id,
+                                 const ov::Tensor& input_ids,
+                                 ov::genai::GenerationConfig sampling_params,
+                                 std::optional<ov::Tensor> token_type_ids = std::nullopt) override;
+
+    GenerationHandle add_request(uint64_t request_id,
+                                 const std::string& prompt,
+                                 ov::genai::GenerationConfig sampling_params) override;
+
+    void set_d2t_for_draft_decoding(std::shared_ptr<ov::op::v0::Constant>& d2t_tensor) {
+        auto eagle_impl = std::dynamic_pointer_cast<ContinuousBatchingForEagleDecodingImpl>(m_draft_pipeline);
+        eagle_impl->set_d2t_for_draft_decoding(d2t_tensor);
```

Comment:
`dynamic_pointer_cast()` requires verification that the result is not null. If you are sure it won't be null, it should be `static_pointer_cast()`